src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -168,18 +181,18 @@ void ColumnBetweenTableScanImpl::_scan_dictionary_segment(
     upper_bound_value_id = segment.unique_values_count();
   }
 
-  const auto value_id_diff = upper_bound_value_id - lower_bound_value_id;
-  const auto comparator = [lower_bound_value_id, value_id_diff](const auto& position) {
-    // Using < here because the right value id is the upper_bound. Also, because the value ids are integers, we can do
-    // a little hack here: (x >= a && x < b) === ((x - a) < (b - a)); cf. https://stackoverflow.com/a/17095534/2204581
-    // This is quite a bit faster.
-    return (position.value() - lower_bound_value_id) < value_id_diff;
-  };
-
-  attribute_vector_iterable.with_iterators(position_filter, [&](auto left_it, auto left_end) {
-    // No need to check for NULL because NULL would be represented as a value ID outside of our range
-    _scan_with_iterators<false>(comparator, left_it, left_end, chunk_id, matches);
-  });
+  with_between_comparator(predicate_condition, lower_bound_value_id, upper_bound_value_id,
```

Comment:
I think the value IDs were already chosen w.r.t. lower/upper exclusive (see L101-L112). Please double-check!

From what I get from the lines above, it might be just be fine to pass `PredicateCondition::BetweenUpperExclusive` here because we adjusted the ValueIDs before s.t. `lower_bound_value_id` is inclusive and `upper_bound_value_id` is exclusive. There might also be an issue with the current solution if the predicate is inclusive, the segment contains NULLs, and the upper bound is higher than the segment maximum. I think we could get around some of the code above, but please to have a look at that.

Reply:
You are right, there were some tests that failed because I used the predicate condition instead of `BetweenUppereExclusive`. I don't think that there will be any other issues, as the code is now doing the exact same thing (just treating the ids as ints and doing the same comparison as before, it's just being routed through `with_between_comparator`)